### ISSUE TITLE
[FEATURE][FormTribe 🔥][500 Points] Add Error handling modal for Link & in-App Surveys

### ISSUE THREAD

**Issue by jobenjada** [formbricks/formbricks#1082]
### Welcome to the FormTribe

This issue will get you the points mentioned in issue title for the [FormTribe Hackathon](https://formtribe.com)

**Preparation:**
If this is your first time contributing, please read:

- Our [introductory blog post](https://formbricks.com/blog/join-the-formtribe)
- Overview of [Code Guidelines at Formbricks.](https://formbricks.com/docs/contributing/how-we-code)

**Only when your PR follows _all_ of the best practices it will get merged.**

Want more points quick? Complete [Side Quests 🏰 ](https://formbricks.com/formtribe#side-quests)

--------

### Is your feature request related to a problem? Please describe.

We introduced optimistic response updates where we queue responses and try to send them in the background without waiting with every question submit.

We retry to resend all response updates up to three times when an error occurs. But when a user finishes the survey and we are still unable to send the responses, the user sees an error modal.

### Describe the solution you'd like

Solution: Please build an error screen that shows all the responses of the user together with the questions plus an option to contact the survey owner.

Tasks

1. In the Team Settings, add a new card above Delete Team:

<img width="710" alt="image" src="https://github.com/formbricks/formbricks/assets/72809645/9e92b2e6-9c6f-4fd6-8a66-03ba3a6fb0a1">
<img width="753" alt="image" src="https://github.com/formbricks/formbricks/assets/72809645/c2cfaa19-a674-4bc7-ae52-bf767a2c1c38">

2. We'll build an error component which is displayed, when the submission failed. To do so, add a new element to the survey package at packages/surveys/components. The component looks as follows:

<img width="627" alt="image" src="https://github.com/formbricks/formbricks/assets/72809645/12a7eff3-d17a-479d-be3f-7e9e386d5859">

- The UI should be self-explanatory :) The Copy your feedback button copies the feedback in a format which makes sense and the Send via Email button opens a mailto: link with the public support email as the recipient
- If the public support email is not set, the "&" and the "Send via email" button isn't displayed.

### UX Notes

- For in-app surveys, we want the modal to close while the data is attempted to submit. Only after it failed a third time, we reopen the modal and display the error component.
- For link surveys, we'll show the Thank You card and replace it with the error component, once the error is thrown.

**Comment by ty-kerr** [comment]
dibs!

**Comment by prasoonmhwr** [comment]
May I work in this

**Comment by anjy7** [comment]
Hey @ty-kerr , are you still working on this ? If not then I would like to :)

**Comment by ty-kerr** [comment]
@anjy7 yeah its in progress now. Should be done tomorrow.
